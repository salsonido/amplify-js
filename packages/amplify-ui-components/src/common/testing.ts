import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
