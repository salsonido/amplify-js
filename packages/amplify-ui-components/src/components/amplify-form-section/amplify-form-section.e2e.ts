import { newE2EPage, E2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-form-section e2e:', () => {
	let page: E2EPage;

	beforeEach(async () => {
		page = await newE2EPage();
		await page.setContent(`<amplify-form-section></amplify-form-section>`);
	});

	it('should render amplify-form-section with a button text of `Submit`', async () => {
		const el = await page.find('amplify-form-section');
		expect(el).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-form-section', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});

	it('should change `button label` to `Enter`', async () => {
		const formSectionEl = await page.find('amplify-form-section');
		expect(formSectionEl).not.toBeNull();

		formSectionEl.setProperty('button-label', 'Enter');

		await page.waitForChanges();

		const buttonLabel = await formSectionEl.getProperty('button-label');

		expect(buttonLabel).toEqual('Enter');
	});

	it('should trigger a submit when `Submit` button is clicked', async () => {
		const formSectionEl = await page.find('amplify-form-section');
		const mockFunc = jest.fn();
		await page.exposeFunction('exposedFunc', mockFunc);

		await page.$eval('amplify-form-section', (componentElement: any) => {
			componentElement.handleSubmit = this.exposedFunc;
		});
		await page.waitForChanges();

		const button = await formSectionEl.find('button');
		expect(mockFunc).not.toHaveBeenCalled();

		await button.click();

		expect(mockFunc).toHaveBeenCalledTimes(1);
	});
});
