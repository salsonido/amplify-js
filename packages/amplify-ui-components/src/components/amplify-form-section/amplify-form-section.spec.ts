import { newSpecPage } from '@stencil/core/testing';
import { AmplifyFormSection } from './amplify-form-section';

describe('amplify-form-section spec:', () => {
	describe('Component logic ->', () => {
		let formSection;

		beforeEach(() => {
			formSection = new AmplifyFormSection();
		});

		it('should render `Submit` for the button text', () => {
			expect(formSection.submitButtonText).toEqual('Submit');
		});
	});
	describe('Render logic ->', () => {
		it('should render a form section with a submit button text of `Go`', async () => {
			const page = await newSpecPage({
				components: [AmplifyFormSection],
				html: `<amplify-form-section submit-button-text="Go"></amplify-form-section>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render a form section with a default button of `Submit`', async () => {
			const page = await newSpecPage({
				components: [AmplifyFormSection],
				html: `<amplify-form-section></amplify-form-section>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render form section', async () => {
			const page = await newSpecPage({
				components: [AmplifyFormSection],
				html: `<amplify-form-section></amplify-form-section>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
