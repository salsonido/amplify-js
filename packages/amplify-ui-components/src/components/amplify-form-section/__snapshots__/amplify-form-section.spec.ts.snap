// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-form-section spec: Render logic -> should render a form section with a default button of \`Submit\` 1`] = `
<amplify-form-section>
  <!---->
  <form>
    <amplify-section>
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="form-section-header">
            <h3 class="header" data-test="form-section-header-section"></h3>
            <div class="subtitle"></div>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="form-section-footer">
            <amplify-button data-test="form-section-form-section-button" type="submit">
              <span>
                Submit
              </span>
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;

exports[`amplify-form-section spec: Render logic -> should render a form section with a submit button text of \`Go\` 1`] = `
<amplify-form-section submit-button-text="Go">
  <!---->
  <form>
    <amplify-section>
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="form-section-header">
            <h3 class="header" data-test="form-section-header-section"></h3>
            <div class="subtitle"></div>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="form-section-footer">
            <amplify-button data-test="form-section-form-section-button" type="submit">
              <span>
                Go
              </span>
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;

exports[`amplify-form-section spec: Render logic -> should render form section 1`] = `
<amplify-form-section>
  <!---->
  <form>
    <amplify-section>
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="form-section-header">
            <h3 class="header" data-test="form-section-header-section"></h3>
            <div class="subtitle"></div>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="form-section-footer">
            <amplify-button data-test="form-section-form-section-button" type="submit">
              <span>
                Submit
              </span>
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;
